autofix-go/string-index-to-contains.batch.yml

```diff
@@ -0,0 +1,22 @@
+name: string-index-to-contains
+description: |
+  This batch change uses [Comby](https://comby.dev) to use String.Contains to check for a substring in a string instead of using Index.
+
+on:
+  - repositoriesMatchingQuery: language:go not file:test not file:vendor not file:Godeps strings.Index(..., ...) > -1 or strings.Index(..., ...) >= 0 or strings.Index(..., ...) != -1 patternType:structural count:100
+
+steps:
+  - run: |
+      comby -in-place 'strings.Index(:[a], :[b]) > -1' 'strings.Contains(:[a], :[b])' .go -matcher .go -exclude-dir .,vendor;
```

Comment:
Probably you can just do `strings.Index(:[a]) > -1` -> `strings.Contains(:[a])`. There's a danger here though: the match pattern will match `strings.Index(:[a]) > -10` or `> -100` or whatever, and you end up replacing to `strings.Contains(:[a])0` or `strings.Contains(:[a])00` or something invalid.

There are a couple of ways to deal with this, but none are super satisfying:

- ensure that there's a space after the number. This means you will miss some expressions inside parens though, like `(blah > 1)`. IMO, nothing to be done about that, and we should just not bother.

- make this apply only for simpler expression that is guaranteed to be valid:

```
if strings.Index(:[a]) > -1 {:[body]}
```

This ensures that you're only modifying an entire if-statement, and it's guaranteed to be valid.

In general I think the above pattern is a bit too eager and is going to trigger false changes, so I would make it more strict.